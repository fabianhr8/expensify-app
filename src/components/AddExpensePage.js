import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';

// This is exported as an unconnected version only for testing
export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startAddExpense(expense);
        this.props.history.push('/');           // Redirect now to the dashboard
    };
    render() {
        return (
            <div>
              <div className='page-header'>
                <div className='content-container'>
                  <h1 className='page-header__title'>Add expense</h1>
                </div>
              </div>
              <div className='content-container'>
                <ExpenseForm
                    onSubmit={this.onSubmit}
                />
              </div>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

// Use connect so we can add the new expense to the store. mapStateToProps is not used here so it is undefined
export default connect(undefined, mapDispatchToProps)(AddExpensePage);
