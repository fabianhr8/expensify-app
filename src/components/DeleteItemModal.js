import React from 'react';

const DeleteItemModal = ({ handleClose, show, handleOnRemove }) => {

  const showHideClassName = show ? "modal display-block" : "modal display-none";

  return (
    <div className={showHideClassName}>
      <section className="modal-main">
        <div className='modal-message'>
          <h3>Do you really want to remove the item?</h3>
        </div>
        <div className='modal-buttons'>
          <div className='modal-button__remove'>
            <button type="button" className='button' onClick={handleOnRemove}>
              Remove item
            </button>
          </div>
          <div className='modal-button__close'>
            <button type="button" className='button' onClick={handleClose}>
              Close
            </button>
          </div>
        </div>
      </section>
    </div>
  );
};

export default DeleteItemModal;
