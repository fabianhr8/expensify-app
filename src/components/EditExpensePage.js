import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, removeExpense, startRemoveExpense } from '../actions/expenses';

// This is exported as an unconnected version only for testing
export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');                    // Redirect to the dashboard
    };
    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');                    // Redirect to the dashboard
    };
    render() {
        return (
            <div>
              <div className='page-header'>
                <div className='content-container'>
                  <h1 className='page-header__title'>Edit expense</h1>
                </div>
              </div>
              <div className='content-container'>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button className='button__delete' onClick={this.onRemove}>Remove Expense</button>
              </div>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => ({
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

// Call the connect() function to find the specific expense to be edited
export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
