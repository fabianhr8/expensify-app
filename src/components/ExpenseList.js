import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

// This is exported as an unconnected version only for testing
export const ExpenseList = (props) => {

    return (
        <div>
            {
                props.expenses.length === 0 ? (
                    <p>There are no expenses</p>
                ) : (
                    props.expenses.map((expense) => {
                        return <ExpenseListItem key={expense.id} {...expense} />
                    })
                )
            }
        </div>
    )};

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters),
    };
}

// We need to call the connect() function with the component ExpenseList to get state from the store
export default connect(mapStateToProps)(ExpenseList);