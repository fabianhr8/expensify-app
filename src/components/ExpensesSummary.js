import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const total = numeral(expensesTotal).format('$0,0.00');
    return (
      <div className='page-header'>
        <div className='content-container'>
          <h1 className='page-header__title'>
            Viewing <span>{expensesCount}</span> {expenseWord} with a total of <span>{total}</span>
          </h1>
          <div className='page-header__actions'>
            <Link className='button' to='/create'>Add Expense</Link>
          </div>
        </div>
      </div>
    )

}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
}

// We need to call the connect() function with the component ExpenseList to get state from the store
export default connect(mapStateToProps)(ExpensesSummary);
