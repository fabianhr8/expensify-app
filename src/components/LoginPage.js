import React from 'react';
import { connect } from 'react-redux';
import { startGoogleLogin, startFacebookLogin } from '../actions/auth';

export const LoginPage = ({ startGoogleLogin, startFacebookLogin }) => (
    <div className='box-layout'>
      <div className='box-layout__box'>
        <h1 className='box-layout__title'>Expensify</h1>
        <p>Get your expenses under control</p>
        <button className='button' onClick={startGoogleLogin}>Log in with Google</button>
        <button className='button' onClick={startFacebookLogin}>Log in with FB</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
  startGoogleLogin: () => dispatch(startGoogleLogin()),
  startFacebookLogin: () => dispatch(startFacebookLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage) ;
