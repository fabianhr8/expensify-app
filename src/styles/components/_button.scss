.button {
  padding: $s-size;
  color: white;
  display: inline-block;
  background-color: $blue;
  border: none;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  text-decoration: none;
  margin-bottom: $s-size;
}

.button--link {
  background: none;
}

.button__delete {
  @extend .button;
  background: #888888;
}
