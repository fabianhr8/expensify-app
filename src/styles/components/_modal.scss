.modal {
  position: fixed;
  top: 0;
  left: 0;
  width:100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);
}

.modal-main {
  position:fixed;
  background: white;
  width: 80%;
  height: auto;
  top:50%;
  left:50%;
  transform: translate(-50%,-50%);
}

.modal-message {
  display: flex;
  justify-content: center;
  margin: $s-size 0;
}

.modal-buttons {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.modal-button__remove {
  margin: $s-size;
}

.modal-button__close {
  margin: $s-size;
}

.display-block {
  display: block;
}

.display-none {
  display: none;
}
