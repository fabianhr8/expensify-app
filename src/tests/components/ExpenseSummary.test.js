import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should render ExpenseSummary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={250}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Should render ExpenseSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={8} expensesTotal={250240}/>);
    expect(wrapper).toMatchSnapshot();
});