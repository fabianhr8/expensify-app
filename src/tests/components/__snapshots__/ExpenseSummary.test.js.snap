// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        8
      </span>
       
      expenses
       with a total of 
      <span>
        $250,240.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should render ExpenseSummary with one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       with a total of 
      <span>
        $250.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
